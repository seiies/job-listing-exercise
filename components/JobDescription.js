import React from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import Paper from 'material-ui/Paper'

const JobDescription = props => {
  const {
    job: {
      title,
      html
    }
  } = props

  return (
    <React.Fragment>
      { /* language=CSS */ }
      <style jsx>{`
        .inner {
          padding: 2rem;
        }
        .back-link, .back-link:visited {
          font-size: 1.3rem;
          color: black;
          text-decoration: none;
        }
        .icon {
          vertical-align: 2px;
        }
        .full-text {
          line-height: 1.4;
        }
      `}</style>

      <Paper className={'paper'}>
        <div className='inner'>
          <Link href='/'>
            <a className='back-link'><span className='icon'>‚Üê</span> Back</a>
          </Link>
          <main>
            <h1>{title}</h1>
            <article className='full-text' dangerouslySetInnerHTML={{__html: html}} />
          </main>
        </div>
      </Paper>
    </React.Fragment>
  )
}

JobDescription.propTypes = {
  className: PropTypes.string,
  job: PropTypes.shape({
    html: PropTypes.string,
    title: PropTypes.string
  }).isRequired
}

JobDescription.defaultProps = {}

export default JobDescription
