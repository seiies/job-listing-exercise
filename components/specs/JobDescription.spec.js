/* global describe, it, expect */

import React from 'react'
import JobDescription from '../JobDescription'
import { shallow } from 'enzyme'

describe('JobDescription', () => {
  let job = {
    'html': 'Get this awesome job',
    'title': 'Job title'
  }

  it('renders correctly', () => {
    const component = shallow(<JobDescription job={job} />)
    expect(component).toMatchSnapshot()
  })
})
