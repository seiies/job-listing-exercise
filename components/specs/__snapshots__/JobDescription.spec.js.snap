// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDescription renders correctly 1`] = `
<React.Fragment>
  <JSXStyle
    css=".inner.jsx-369923791{padding:2rem;}.back-link.jsx-369923791,.back-link.jsx-369923791:visited{font-size:1.3rem;color:black;-webkit-text-decoration:none;text-decoration:none;}.icon.jsx-369923791{vertical-align:2px;}.full-text.jsx-369923791{line-height:1.4;}"
    styleId="369923791"
  />
  <WithStyles(Paper)
    className="paper"
  >
    <div
      className="jsx-369923791 inner"
    >
      <Link
        href="/"
      >
        <a
          className="jsx-369923791 back-link"
        >
          <span
            className="jsx-369923791 icon"
          >
            ‚Üê
          </span>
           Back
        </a>
      </Link>
      <main
        className="jsx-369923791"
      >
        <h1
          className="jsx-369923791"
        >
          Job title
        </h1>
        <article
          className="jsx-369923791 full-text"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Get this awesome job",
            }
          }
        />
      </main>
    </div>
  </WithStyles(Paper)>
</React.Fragment>
`;
